src/client/mod.rs

```diff
@@ -17,7 +17,10 @@ use token::{Token, Lifetime, Refresh};
 
 /// OAuth 2.0 client.
 #[derive(Debug, Clone, PartialEq, Eq)]
-pub struct Client<P: Provider> {
+pub struct Client<P> {
```

Comment:
Why drop the Provider trait constraint on P here? In what circumstances would a Client<P> not have P impl Provider?